File: Data API/Websocket/Javascript/ws.js
// import { EventEmitter } from "events";
// const EventEmitter = require("events")

// export default class WsClient extends EventEmitter {
class WsClient {
  constructor(host, port, type) {

    this.server = {
      host,
      port,
      type
    };

    this.settings = {
      reconnect: true,
      retry: 3000,
      decode: true
      // data: 'all'
    };

    this.connect()
  }
  connect() {
    console.log(
      "Initiating connection to " +
        this.server.host +
        " " +
        this.server.port
    );

    this.connection = new WebSocket(
      this.server.type +
        "://" +
        this.server.host +
        ":" +
        this.server.port
    );

    this.connection.addEventListener("open", this);

    this.connection.addEventListener("close", this);

    this.connection.addEventListener("message", this);
  }
  handleEvent(evt) {
    switch (evt.type) {
      case "message":
        console.log('MSG:')
        this.decode(evt.data)
        break;
      case "open":
        this.state = WebSocket.OPEN;
        // this.emit("open");
        console.log("Connection established");
        break;
      case "close":
        console.log("Connection closed");
        this.connection.removeEventListener("open", this);
        this.connection.removeEventListener("close", this);
        this.connection.removeEventListener("message", this);

        // this.emit("close");
        break;
      default:
        console.log("Unhandled evt");
    }
  }

  decode(data) { 
    let parsed
    if (data.substr(0, 3) == "cx|") {
      parsed = JSON.parse(data.slice(10));
    } else if (data.charAt(0) === "{") {
      parsed = JSON.parse(data);
    } else {
      console.log('Received: ' + data)
    }

    if ('frames' in parsed) {
      console.log('POSXYZ');
    } else if ('anchorList' in parsed) {
      console.log('ALIST')
    } else if ('tagList' in parsed) {
      console.log('TLIST')
    }
    else if ('tagStatus' in parsed) {
      console.log('Status')
    } else {
      console.log('UNKNOWN PKT')
    }
    console.log(parsed)
  }

}

// module.exports = WsClient

File: Data API/Websocket/Python/ws.py
import asyncio
import json
import pprint
import websockets

async def connect(
    type_:str = 'ws',
    host:str = 'localhost',
    port: int = 13510
) -> None:
    """Open a WebSocket connection to the server, handling incoming messages.

    Parameters are used to form a URI like 'ws://localhost:1234'.

    ## Parameters
        type_: Type of the connection.
        host: Hostname.
        port: Port.

    ## Returns
        None.
    """
    uri = '{}://{}:{}'.format(type_, host, port)
    print('Initiating connection to {}'.format(uri))

    async with websockets.connect(uri) as websocket:
        if websocket.open:
            print('Connection established')
        async for message in websocket:
            await process_message(message)
    if websocket.closed:
        print('Connection closed')

async def process_message(message: str) -> None:
    """Parse an incoming message, printing it's type and itself.

    ## Parameters
        message:

    ## Returns
        None.
    """
    if 'cx|' in message:
        parsed = json.loads(message[10:])
    elif message[0] == '{':
        parsed = json.loads(message)
    else:
        print('Received: {}'.format(message))

    if 'frames' in parsed:
        print('POSXYZ')
    elif 'anchorList' in parsed:
        print('ALIST')
    elif 'tagList' in parsed:
        print('TLIST')
    elif 'tagStatus' in parsed:
        print('Status')
    else:
        print('UNKNOWN PKT')
    
    pprint.PrettyPrinter(indent=2).pprint(parsed)

if __name__ == '__main__':
    type_ = 'ws'
    host = 'localhost'
    port = 13510

    asyncio.run(connect(type_, host, port))